src/headers/mod.rs

```diff
@@ -0,0 +1,43 @@
+use crate::error::VssError;
+use async_trait::async_trait;
+use reqwest::header::HeaderMap;
+use std::collections::HashMap;
+use std::str::FromStr;
+
+/// Defines a trait around how headers are provided for each VSS request.
+#[async_trait]
+pub trait VssHeaderProvider {
+	/// Returns the HTTP headers to be used for a VSS request.
+	/// This method is called on each request, and should likely perform some form of caching.
+	async fn get_headers(&self, request: &[u8]) -> Result<HashMap<String, String>, VssError>;
+}
+
+/// A header provider returning an given, fixed set of headers.
+pub struct FixedHeaders {
+	headers: HashMap<String, String>,
+}
+
+impl FixedHeaders {
+	/// Creates a new header provider returning the given, fixed set of headers.
+	pub fn new(headers: HashMap<String, String>) -> FixedHeaders {
+		FixedHeaders { headers }
+	}
+}
+
+#[async_trait]
+impl VssHeaderProvider for FixedHeaders {
+	async fn get_headers(&self, _request: &[u8]) -> Result<HashMap<String, String>, VssError> {
```

Comment:
Yeah I agree, this was the main reason I initially created a separate error type. Moved to `io::Result` which will be embedded in `VssError::AuthError`.

Reply:
Ah right, I forgot whether this will turn out internal in `ldk-node` or exposed as bindings. We did discuss the option of having this exposed as bindings, so I changed this into `Result<_, String>` to not have different semantics around `VssError` for different cases.

Reply:
Ah, thanks for mentioning that! I should have double-checked UniFFI. Reverted back to `VssError` and added documentation in the trait. Enforced all VssHeaderProvider errors to be a single variant i.e. `VssError::AuthError` to not conflate semantics.

Reply:
Reverted back to the original UniFFI compatible error type previous to https://github.com/lightningdevkit/vss-rust-client/pull/26#discussion_r1600472044.